packages/backend/src/services/searchService/integrationTests/utilities/requestBuilder/buildRegexp.test.js

```diff
@@ -2,8 +2,195 @@ const { describe: unitDescribe } = require('common/src/testUtilities/envUnit')
 
 const buildRegexp = require('./buildRegexp')
 
-unitDescribe('searchSpecimen - query - tmp - buildRegexp', () => {
-  it('works', () => {
-    expect(buildRegexp('test')).toBe('test')
+const runTestCases = testCases => {
+  testCases.forEach(testCase => {
+    const { input, matching = [], notMatching = [], errorMessage } = testCase
+    describe(input, () => {
+      if (errorMessage) {
+        it(`${input} should throw: ${errorMessage || ''}`, () => {
+          expect(() => buildRegexp(input)).toThrow(errorMessage)
+        })
+      } else {
+        const regexpString = buildRegexp(input)
+        matching.forEach(string => {
+          const regexp = new RegExp(regexpString)
```

Comment:
maybe could be done outside of forEach?

Reply:
The regular expression have memory so I want a new one for each test

Reply:
ah, good